logic/extensions/atrium-logic-kotlin_1_3/src/commonMain/kotlin/ch/tutteli/atrium/logic/kotlin_1_3/impl/DefaultResultAssertions.kt

```diff
@@ -29,16 +26,36 @@ class DefaultResultAssertions : ResultAssertions {
             .withoutOptions()
             .build()
 
+
+
         @OptIn(ExperimentalNewExpectTypes::class, ExperimentalComponentFactoryContainer::class)
     override fun <TExpected : Throwable> isFailureOfType(
-        container: AssertionContainer<out Result<*>>,
-        expectedType: KClass<TExpected>
-    ):  SubjectChangerBuilder.ExecutionStep<Throwable?, TExpected> {
+            container: AssertionContainer<out Result<*>>,
+            expectedType: KClass<TExpected>) : SubjectChangerBuilder.ExecutionStep<Throwable?, TExpected> {
+
+
         println("heeere and subject is: ${container.maybeSubject.getOrElse { "no subject?" }}")
-        return container.manualFeature(EXCEPTION) {
-            println("suddenly no longer a Failure?: $this")
+
+        return container.manualFeature(EXCEPTION) {println("suddenly no longer a Failure?: $this")
             exceptionOrNull()
```

Comment:
check needs to be in here and you need to check, if exceptionOrNull() != null and then check that `container.maybeSubject.map { exceptionOrNull() }` was not null as well